website/components/AppWrapper.tsx

```diff
@@ -62,17 +62,24 @@ export default function AppWrapper(inputProps: AppWrapperProps): ReactElement {
   const location = useLocation();
   const navigation = useNavigate();
 
-  const [viewerSettings, setViewerSettings] = useState<Partial<GlobalViewerSettings>>(props.viewerSettings);
-  const [viewerArgs, setViewerArgs] = useState<AppDataProps>(props.viewerArgs);
+  const [viewerSettings, setViewerSettings] = useState<Partial<GlobalViewerSettings> | null>(null);
+  const [viewerArgs, setViewerArgs] = useState<AppDataProps | null>(null);
```

Comment:
it's confusing to me that there are props.viewerArgs and this state.viewerArgs. Where are the props.viewerArgs coming from? Also since you have viewerSettings, maybe `viewerProps` is a better name for `viewerArgs`? 

If viewerArgs and viewerSettings are getting derived once from a combination of props and url params, it seems like you don't need `useState` at all since they're never getting changed

Reply:
That rename makes sense to me, especially since the underlying type for `viewerArgs` is `AppDataProps`. `viewerArgs` is usually being derived from the browser's location (data passed as part of the navigation state) or from the URL. I think I'll remove the `prop` for it since `AppWrapper` is only used on the website and doesn't really have parent components that would need to pass in prop data.

I think I would still need a `useState` since the derivation from the URL is asynchronous... Can you let me know if you see another way to do it?

Reply:
one thought is just to use `useRef` for the actual data and then store a single `requestHasFinished` flag in state instead of two separate state updates. But since these weren't added by you in this PR and you're fixing a bug I think it makes sense to leave them as is for now. But a refactor where you're not storing props in state and then exporting them as props again I think would be good